packages/@aws-cdk/aws-efs/lib/efs-file-system.ts

```diff
@@ -231,37 +217,32 @@ export class EfsFileSystem extends EfsFileSystemBase {
    */
   public readonly fileSystemId: string;
 
-  private readonly efsFileSystem: CfnFileSystem;
-
   /**
    * Constructor for creating a new EFS FileSystem.
    */
-  constructor(scope: Construct, id: string, props: EfsFileSystemProps) {
+  constructor(scope: Construct, id: string, props: FileSystemProps) {
     super(scope, id);
 
-    if (props.throughputMode === EfsThroughputMode.PROVISIONED) {
-      if (props.provisionedThroughputInMibps === undefined) {
+    if (props.throughputMode === ThroughputMode.PROVISIONED) {
+      if (props.provisionedThroughputPerSecond === undefined) {
         throw new Error('Property provisionedThroughputInMibps is required when throughputMode is PROVISIONED');
-      } else if (!Number.isInteger(props.provisionedThroughputInMibps)) {
-        throw new Error('Invalid input for provisionedThroughputInMibps');
-      } else if (props.provisionedThroughputInMibps < 1 || props.provisionedThroughputInMibps > 1024) {
-        this.node.addWarning('Valid values for throughput are 1-1024 MiB/s. You can get this limit increased by contacting AWS Support.');
+      } else if (props.provisionedThroughputPerSecond.toMebibytes() > 1024) {
+        throw new Error('Valid values for throughput are 1MiB/s - 1GiB/s. You can get this limit increased by contacting AWS Support.');
```

Comment:
does this mean this can be increased beyond 1GiB/s through AWS support? won't this validation stand in the way for users who have it increased?

Reply:
:D good catch

Reply:
I'm just cleaning up what was already there.

I'll fix this up.

Reply:
@nija-at it is turning a warning into an Error no?

Reply:
Yes. Seems like the right thing to do here.